Neos.Fusion/Classes/Core/Runtime.php

```diff
@@ -179,25 +222,47 @@ public function addCacheTag($key, $value)
     /**
      * Completely replace the context array with the new $contextArray.
      *
-     * Purely internal method, should not be called outside of Neos.Fusion.
-     *
+     * @internal purely internal method, should not be called outside Neos.Fusion.
      * @param array $contextArray
      * @return void
      */
     public function pushContextArray(array $contextArray)
     {
+        foreach ($this->fusionGlobals->value as $disallowedFusionContextKey => $_) {
+            if (array_key_exists($disallowedFusionContextKey, $contextArray)) {
+                throw new RuntimeException(sprintf('Overriding Fusion global variable "%s" via @context is not allowed.', $disallowedFusionContextKey), 1694247627130);
+            }
+        }
```

Comment:
@mficzel or is it faster to first iterate over the `$contextArray` -> would be if it has less entries than `$this->fusionGlobals->value`

Reply:
I think it would be way faster to check the keys that were used in `$fusionConfiguration['__meta']['context']` here: 
https://github.com/neos/neos-development-collection/blob/9.0/Neos.Fusion/Classes/Core/Runtime.php#L563C14-L568. Similar to what you did in `pushContext`. That way we check only the newly added context keys and not everything. That would still prevent any global overriding from fusion and `pushContext`but leave  `pushContextArray` open for "creative" use via php. 

`pushContextArray` is described as beeing internal anyways and should get that as official annotation. So i would suggest to only add the check `pushContext` and the handling of `['__meta']['context']` and leave `pushContextArray` as it is as an internal footgun. 

I can can adjust Fusion.Form to set the `request` via `pushContextArray` so we need no special rules there. As this is in the Fusion namespace i consider using internal api fine there.

Reply:
I like this approach 👍

Reply:
Thanks for the rich discussions ^^, its now implemented as you described.